plugins/Insights/Visualizations/Insight.php

```diff
@@ -59,6 +59,17 @@ public function beforeLoadDataTable()
         );
     }
 
+    private function getComparedToXPeriodsAgo()
+    {
+        $period = Common::getRequestVar('period', null, 'string');
```

Comment:
@sgiehl took me a while to understand what is happening here and seeing no `Range::getDateXPeriodsAgo()` is converting eg indeed `365` to 365 months ago. From what I see `week` and `year` only can compare to the previous period, but `compared_to_x_periods_ago` can be `1` or `12` for months. Would we need to convert eg a number `365` to `12` instead of 1 when the period is month?

Also probably not needed but could maybe have a side effect... (probs not and only if plugins where to maybe hook into it somehow or so)... Would we maybe need to change `$this->requestConfig->compared_to_x_periods_ago` in above code and set the updated value to `comparedToXPeriods`? I don't think it's actually used in any way but may be better.

Reply:
added another condition for month to fix that